spring-pulsar-test/src/main/java/org/springframework/pulsar/test/support/PulsarConsumerTestUtil.java

```diff
@@ -53,6 +55,41 @@ public class PulsarConsumerTestUtil<T> implements TopicSpec<T>, SchemaSpec<T>, C
 
 	private List<String> topics;
 
+	private boolean calledUntil = false;
+
+	public static <T> TopicSpec<T> consumeMessages() throws PulsarClientException {
+		PulsarClient pulsarClient;
+		try {
+			pulsarClient = PulsarClient.builder().serviceUrl(
+				"pulsar://localhost:6650"
+			).build();
+		}
+		catch (PulsarClientException e) {
+			pulsarClient = PulsarClient.builder().serviceUrl(
+				PulsarTestContainerSupport.getPulsarBrokerUrl()
```

Comment:
What if the test using `consumeMessages()` does not rely on`PulsarTestContainerSupport`? In that case , I believe, the pulsar container will not have been started. Maybe we should verify if the `PulsarTestContainerSupport` container is actually being used (started) and if not throw a proper exception?
Or even better, this could be the first check at the start of the method? Verify if the container is started, if so use its `serviceUrl` else default to `localhost:6650`?

Reply:
Good point @jonas-grgt , I think flipping them makes sense. 

Also, it would be helpful to add the following to PTCS to ease this check.

```java
static boolean isContainerStarted() {
	return PULSAR_CONTAINER.isRunning();
}
```

With this in place we can heavily assume that if the PTCS is running then this is the only way we should try to connect. It would be confusing if it was running but for some reason it failed to connect and then we try `localhost`. 

Something like this: 

```java
public static <T> TopicSpec<T> consumeMessages() {
	if (!PulsarTestContainerSupport.isContainerStarted()) {
		return PulsarConsumerTestUtil.consumeMessages(PulsarTestContainerSupport.getPulsarBrokerUrl());
	}
	return PulsarConsumerTestUtil.consumeMessages("pulsar://localhost:6650");
}

```